activerecord/test/cases/persistence_test.rb

```diff
@@ -125,7 +125,10 @@ def test_destroy_all
     assert ! topics_by_mary.empty?
 
     assert_difference('Topic.count', -topics_by_mary.size) do
-      destroyed = Topic.destroy_all(conditions).sort_by(&:id)
+      assert_deprecated do
+        Topic.destroy_all(conditions)
+      end
+      destroyed = Topic.where(conditions).destroy_all.sort_by(&:id)
```

Comment:
Won't this change the records returned, since you destroyed them above? We should continue to assign `destroyed` from the `assert_deprecated` (though you'll probably need to assign it to nil outside of the block)